Firestore/core/test/firebase/firestore/immutable/sorted_set_test.cc

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/immutable/sorted_set.h"
+
+#include <random>
+#include <unordered_set>
+
+#include "Firestore/core/test/firebase/firestore/immutable/testing.h"
+
+using firebase::firestore::immutable::impl::SortedMapBase;
+using SizeType = SortedMapBase::size_type;
+
+namespace firebase {
+namespace firestore {
+namespace immutable {
+
+template <typename Container, typename K>
+testing::AssertionResult Found(const Container& container, const K& key) {
+  if (!container.contains(key)) {
+    return testing::AssertionFailure()
+           << "Did not find key " << key << " using contains()";
+  }
+
+  auto found = container.find(key);
+  if (found == container.end()) {
+    return testing::AssertionFailure()
+           << "Did not find key " << key << " using find()";
+  }
+  if (*found == key) {
+    return testing::AssertionSuccess();
+  } else {
+    return testing::AssertionFailure()
+           << "Found entry was " << Describe(*found);
+  }
+}
+
+template <typename K>
+SortedSet<K> ToSet(const std::vector<K>& container) {
+  SortedSet<K> result;
+  for (auto&& entry : container) {
+    result = result.insert(entry);
+  }
+  return result;
+}
+
+static const int kLargeNumber = 100;
+
+TEST(SortedSetTest, EmptyBehavior) {
+  SortedSet<int> set;
+
+  EXPECT_TRUE(set.empty());
+  EXPECT_EQ(0u, set.size());
+
+  EXPECT_TRUE(NotFound(set, 1));
+}
```

Comment:
This test exists in the `SortedMap` tests, so I'll pass here. These tests exist because I wanted a smoke test that this worked independent of actual usage.